### PR TITLE
docs: Document desired crate dependencies in comments

### PR DESCRIPTION
**Pull request by alamb**
This explains the desired dependency structure between storage crates (`mutable_buffer` and `read_buffer`) and the query crate (`query`) as I believe @e-dard  and I would like

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: query/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,16 @@ description = "IOx Query Interface and Executor"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+# This create is designed to be indepndent of the rest of the IOx
```
Comment (e-dard) [inline query/Cargo.toml:11]:
```suggestion
# This create is designed to be independent of the rest of the IOx
```

**Review by pauldix — APPROVED** [review]
LGTM, although I thought an additional desire of the dependency structure is that it would make clear that the query crate has shared functionality so that front ends will use that regardless of which storage backend you're hitting.

File: mutable_buffer/Cargo.toml [hunk 2]
```diff
@@ -4,6 +4,16 @@ version = "0.1.0"
 authors = ["Andrew Lamb <andrew@nerdnetworks.org>"]
 edition = "2018"
 
+
+# Note the long term goal is that this crate is standalone and not
+# depend on the IOx Query Engine (even though this is not the case
```
Comment (e-dard) [inline mutable_buffer/Cargo.toml:9]:
```suggestion
# dependent on the IOx Query Engine (even though this is not the case
```

File: read_buffer/Cargo.toml [hunk 3]
```diff
@@ -4,8 +4,16 @@ version = "0.1.0"
 authors = ["Edd Robinson <me@edd.io>"]
 edition = "2018"
 
+
+
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+# Note this crate is designed to be standalone that should not depend
```
Comment (e-dard) [inline read_buffer/Cargo.toml:11]:
```suggestion
# Note this crate is designed to be standalone, and should not depend
```

**Review by e-dard — DISMISSED** [review]
Couple of typos otherwise LGTM 👍

**Comment by alamb** [comment]
Thank you @e-dard for fixing up my very sloppy writing

**Pull request merged**